rating-service/src/main/java/io/simakkoi9/ratingservice/service/impl/RatingServiceImpl.java

```diff
@@ -0,0 +1,146 @@
+package io.simakkoi9.ratingservice.service.impl;
+
+import io.quarkus.hibernate.orm.panache.PanacheQuery;
+import io.quarkus.panache.common.Page;
+import io.simakkoi9.ratingservice.config.MessageConfig;
+import io.simakkoi9.ratingservice.exception.DriverAlreadyRatedException;
+import io.simakkoi9.ratingservice.exception.DuplicateRatingException;
+import io.simakkoi9.ratingservice.exception.NoRatesException;
+import io.simakkoi9.ratingservice.exception.PassengerAlreadyRatedException;
+import io.simakkoi9.ratingservice.exception.RatingNotFoundException;
+import io.simakkoi9.ratingservice.model.dto.request.DriverRatingUpdateRequest;
+import io.simakkoi9.ratingservice.model.dto.request.PassengerRatingUpdateRequest;
+import io.simakkoi9.ratingservice.model.dto.request.RatingCreateRequest;
+import io.simakkoi9.ratingservice.model.dto.response.AverageRatingResponse;
+import io.simakkoi9.ratingservice.model.dto.response.RatingPageResponse;
+import io.simakkoi9.ratingservice.model.dto.response.RatingResponse;
+import io.simakkoi9.ratingservice.model.entity.Rating;
+import io.simakkoi9.ratingservice.model.mapper.RatingMapper;
+import io.simakkoi9.ratingservice.repository.RatingRepository;
+import io.simakkoi9.ratingservice.service.RatingService;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+
+import java.util.List;
+import java.util.Objects;
+
+@ApplicationScoped
+public class RatingServiceImpl implements RatingService {
+
+    @Inject
+    RatingMapper ratingMapper;
+    @Inject
+    RatingRepository ratingRepository;
+    @Inject
+    MessageConfig messageConfig;
+
+    @Override
+    @Transactional
+    public RatingResponse createRating(RatingCreateRequest ratingCreateRequest) {
+        if (ratingRepository.existsByRideId(ratingCreateRequest.rideId())) {
+            throw new DuplicateRatingException("duplicate.rating", messageConfig, ratingCreateRequest.rideId());
+        }
+        //Нужна еще проверка наличия в сервисе поездок
+        Rating rating = ratingMapper.toEntity(ratingCreateRequest);
+        rating.persist();
+        return ratingMapper.toResponse(rating);
+    }
+
+    @Override
+    @Transactional
+    public RatingResponse setRateForDriver(Long id, DriverRatingUpdateRequest updateRequest) {
+        Rating rating = findRatingByIdOrElseThrow(id);
+        if (rating.getRateForDriver() != null){
+            throw new DriverAlreadyRatedException("driver.already.rated", messageConfig, id);
+        }
+        Rating updatedRating = ratingMapper.driverRatingPartialUpdate(updateRequest, rating);
+
+        return ratingMapper.toResponse(updatedRating);
+    }
+
+    @Override
+    @Transactional
+    public RatingResponse setRateForPassenger(Long id, PassengerRatingUpdateRequest updateRequest) {
+        Rating rating = findRatingByIdOrElseThrow(id);
+        if (rating.getRateForPassenger() != null){
```

Comment:
Здесь и далее - необходим пробел перед открывающей фигурной скобкой:
```suggestion
        if (rating.getRateForPassenger() != null) {
```